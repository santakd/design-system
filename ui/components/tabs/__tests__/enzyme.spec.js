// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import { mount } from 'enzyme';
import MobileTabs from '../mobile-stack';

const { matchesMarkup } = createHelpers(__dirname);

describe('MobileTabs', () => {
  it('handles a click on a tab', () => {
    const wrapper = mount(<MobileTabs />);
    wrapper
      .find('.slds-button')
      .at(0)
      .simulate('click');
    wrapper.update();
    return matchesMarkup(wrapper.html());
  });

  it('adds the open class to the panel when tab is clicked', () => {
    const wrapper = mount(<MobileTabs />);
    wrapper
      .find('.slds-button')
      .at(0)
      .simulate('click');
    wrapper.update();
    expect(
      wrapper
        .find('.slds-panel')
        .at(0)
        .find('.slds-is-open')
    ).toHaveLength(1);
    expect(wrapper.find('.slds-tabs-mobile.slds-hidden')).toHaveLength(1);
  });

  it('handles a click on the back button of a panel', () => {
    const wrapper = mount(<MobileTabs />);
    wrapper
      .find('.slds-button')
      .at(0)
      .simulate('click');
    wrapper.update();
    wrapper
      .find('.slds-panel__back')
      .at(0)
      .simulate('click');
    wrapper.update();

    return matchesMarkup(wrapper.html());
  });

  it('adds the hidden class back to the panel when back button is pressed', () => {
    const wrapper = mount(<MobileTabs />);
    wrapper
      .find('.slds-button')
      .at(0)
      .simulate('click');
    wrapper.update();
    wrapper
      .find('.slds-panel__back')
      .at(0)
      .simulate('click');
    wrapper.update();
    expect(wrapper.find('.slds-panel.slds-hidden')).toHaveLength(3);
  });
});
