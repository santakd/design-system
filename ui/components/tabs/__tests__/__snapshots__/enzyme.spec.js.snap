// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileTabs handles a click on a tab 1`] = `"<div class=\\"slds-tabs-mobile__container\\"><ul class=\\"slds-tabs-mobile slds-hidden\\"><li class=\\"slds-tabs-mobile__item\\"><button class=\\"slds-button slds-button_full-width\\"><span class=\\"slds-truncate\\" title=\\"Item One\\">Item One</span><svg class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#chevronright\\"></use></svg></button></li><li class=\\"slds-tabs-mobile__item\\"><button class=\\"slds-button slds-button_full-width\\"><span class=\\"slds-truncate\\" title=\\"Item Two\\">Item Two</span><svg class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#chevronright\\"></use></svg></button></li><li class=\\"slds-tabs-mobile__item\\"><button class=\\"slds-button slds-button_full-width\\"><span class=\\"slds-truncate\\" title=\\"Item Three\\">Item Three</span><svg class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#chevronright\\"></use></svg></button></li></ul><div class=\\"slds-panel slds-panel_docked slds-panel_docked-left slds-panel_animated slds-is-open\\" aria-hidden=\\"false\\"><div class=\\"slds-panel__header slds-panel__header_align-center\\"><button class=\\"slds-button slds-button_icon slds-button_icon-small slds-panel__back\\" title=\\"Collapse Item 1 Panel\\"><svg class=\\"slds-button__icon\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#back\\"></use></svg><span class=\\"slds-assistive-text\\">Collapse Item 1 Panel</span></button><h2 class=\\"slds-panel__header-title slds-text-heading_small slds-truncate\\" title=\\"Item 1 Panel\\">Item 1 Panel</h2></div><div class=\\"slds-panel__body\\">A panel body accepts any layout or component</div></div><div class=\\"slds-panel slds-panel_docked slds-panel_docked-left slds-panel_animated slds-hidden\\" aria-hidden=\\"true\\"><div class=\\"slds-panel__header slds-panel__header_align-center\\"><button class=\\"slds-button slds-button_icon slds-button_icon-small slds-panel__back\\" title=\\"Collapse Item 2 Panel\\"><svg class=\\"slds-button__icon\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#back\\"></use></svg><span class=\\"slds-assistive-text\\">Collapse Item 2 Panel</span></button><h2 class=\\"slds-panel__header-title slds-text-heading_small slds-truncate\\" title=\\"Item 2 Panel\\">Item 2 Panel</h2></div><div class=\\"slds-panel__body\\">A panel body accepts any layout or component</div></div><div class=\\"slds-panel slds-panel_docked slds-panel_docked-left slds-panel_animated slds-hidden\\" aria-hidden=\\"true\\"><div class=\\"slds-panel__header slds-panel__header_align-center\\"><button class=\\"slds-button slds-button_icon slds-button_icon-small slds-panel__back\\" title=\\"Collapse Item 3 Panel\\"><svg class=\\"slds-button__icon\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#back\\"></use></svg><span class=\\"slds-assistive-text\\">Collapse Item 3 Panel</span></button><h2 class=\\"slds-panel__header-title slds-text-heading_small slds-truncate\\" title=\\"Item 3 Panel\\">Item 3 Panel</h2></div><div class=\\"slds-panel__body\\">A panel body accepts any layout or component</div></div></div>"`;

exports[`MobileTabs handles a click on the back button of a panel 1`] = `"<div class=\\"slds-tabs-mobile__container\\"><ul class=\\"slds-tabs-mobile\\"><li class=\\"slds-tabs-mobile__item\\"><button class=\\"slds-button slds-button_full-width\\"><span class=\\"slds-truncate\\" title=\\"Item One\\">Item One</span><svg class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#chevronright\\"></use></svg></button></li><li class=\\"slds-tabs-mobile__item\\"><button class=\\"slds-button slds-button_full-width\\"><span class=\\"slds-truncate\\" title=\\"Item Two\\">Item Two</span><svg class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#chevronright\\"></use></svg></button></li><li class=\\"slds-tabs-mobile__item\\"><button class=\\"slds-button slds-button_full-width\\"><span class=\\"slds-truncate\\" title=\\"Item Three\\">Item Three</span><svg class=\\"slds-icon slds-icon_x-small slds-icon-text-default\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#chevronright\\"></use></svg></button></li></ul><div class=\\"slds-panel slds-panel_docked slds-panel_docked-left slds-panel_animated slds-hidden\\" aria-hidden=\\"true\\"><div class=\\"slds-panel__header slds-panel__header_align-center\\"><button class=\\"slds-button slds-button_icon slds-button_icon-small slds-panel__back\\" title=\\"Collapse Item 1 Panel\\"><svg class=\\"slds-button__icon\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#back\\"></use></svg><span class=\\"slds-assistive-text\\">Collapse Item 1 Panel</span></button><h2 class=\\"slds-panel__header-title slds-text-heading_small slds-truncate\\" title=\\"Item 1 Panel\\">Item 1 Panel</h2></div><div class=\\"slds-panel__body\\">A panel body accepts any layout or component</div></div><div class=\\"slds-panel slds-panel_docked slds-panel_docked-left slds-panel_animated slds-hidden\\" aria-hidden=\\"true\\"><div class=\\"slds-panel__header slds-panel__header_align-center\\"><button class=\\"slds-button slds-button_icon slds-button_icon-small slds-panel__back\\" title=\\"Collapse Item 2 Panel\\"><svg class=\\"slds-button__icon\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#back\\"></use></svg><span class=\\"slds-assistive-text\\">Collapse Item 2 Panel</span></button><h2 class=\\"slds-panel__header-title slds-text-heading_small slds-truncate\\" title=\\"Item 2 Panel\\">Item 2 Panel</h2></div><div class=\\"slds-panel__body\\">A panel body accepts any layout or component</div></div><div class=\\"slds-panel slds-panel_docked slds-panel_docked-left slds-panel_animated slds-hidden\\" aria-hidden=\\"true\\"><div class=\\"slds-panel__header slds-panel__header_align-center\\"><button class=\\"slds-button slds-button_icon slds-button_icon-small slds-panel__back\\" title=\\"Collapse Item 3 Panel\\"><svg class=\\"slds-button__icon\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#back\\"></use></svg><span class=\\"slds-assistive-text\\">Collapse Item 3 Panel</span></button><h2 class=\\"slds-panel__header-title slds-text-heading_small slds-truncate\\" title=\\"Item 3 Panel\\">Item 3 Panel</h2></div><div class=\\"slds-panel__body\\">A panel body accepts any layout or component</div></div></div>"`;
